pkg/sandboxes/aws-ecs-byovpc/cloudformation-template-delegation.yaml

```diff
@@ -0,0 +1,70 @@
+# generated on: {{ .GeneratedOn }}
+Parameters:
+  RoleName:
+    Type: String
+    Default: "nuon-aws-ecs-byovpc-install-access"
+    Description: "Enter a name for the install role"
+  DelegationRoleARN:
+    Type: String
+    Default: ""
+    Description: "ARN for the role that will be assumed by roles in the trust policy."
+
+Outputs:
+  RoleARN:
+    Description: IAM Role ARN
+    Value: !GetAtt InstallRole.Arn
+
+Resources:
+  # Macro:
+  #   Type: "AWS::CloudFormation::Macro"
+  #   Properties:
+  #     Description: 'Macro for generating a list of entries for a policy.'
+  #     FunctionName: 'arn:aws:lambda:us-west-2:949309607565:function:NuonCFTransformTemplateECSDelegated'
+  #     Name: NuonCFTransformTemplateECSDelegated
+
+  InstallRole:
+    Type: AWS::IAM::Role
+    Properties:
+      RoleName:
+        Ref: RoleName
+      Tags:
+        - Key: "managed-by"
+          Value: "nuon"
+        - Key: "nuon-sandbox-type"
+          Value: "aws-ecs-byovpc"
+      AssumeRolePolicyDocument:
+        Statement:
+          - Effect: Allow
+            Action: sts:AssumeRole
+            Principal:
+              AWS: !Ref DelegationRoleARN
+
+  ProvisionPolicy:
+    Type: AWS::IAM::Policy
+    Properties:
+      PolicyName: !Sub
+        - nuon-provision-${RoleName}
+        - RoleName: !Ref RoleName
+      Roles:
+        - !Ref InstallRole
```

Comment:
yeah. i'm actually treating these as plain go templates and not using the yaml parser to avoid any potential issues. these are expanded when they are executed, so it shouldn't be an issue for us; but i don't want any surprises.